nabs/preprocessors.py

```diff
@@ -301,4 +302,70 @@ def daq_during_plan():
     return (yield from daq_during_plan())
 
 
+def step_size_decorator(plan):
+    """
+    Grab the last argument (number of steps), and intepret as
+    - step size if float
+    - number of steps if integer
+
+    Only works on step scans in one dimension.
+
+    Parameters
+    ----------
+    plan : plan
+        A bluesky plan that yields bluesky Msg objects.  Must be
+        a scan in one dimension, with the last argument being
+        the number of scan points / step size.
+
+    Returns
+    -------
+    step_size_plan : plan
+        The same plan as before, but modified appropriately to
+        differentiate between step size and number of steps.
+        This will be a callable generator function.
+    """
+
+    @wraps(plan)
+    def inner(*args, **kwargs):
+        if 'num' in kwargs:
+            # Currently unneeded, since daq_ascan and daq_dscan
+            # do not support num kwarg
+            n = kwargs.pop('num')
+        else:
+            # assumes (det_list, motor, start, stop, num)
+            n = args[-1]
+
+        if not isinstance(n, (int, float)):
+            raise TypeError("Step size / number of steps is "
+                            "neither float nor integer")
+
+        if type(n) is int:
```

Comment:
This is likely excessively cautious, but it's not impossible that the scientists would get step counts/step sizes from numpy calculations which don't return native floats/ints:

An alternative could be, e.g.:

```python
In [3]: isinstance(1, numbers.Integral)
Out[3]: True

In [4]: isinstance(1., numbers.Integral)
Out[4]: False

In [5]: import numpy as np

In [6]: isinstance(np.uint(8), numbers.Integral)
Out[6]: True
```

Otherwise falling back to `step_size = float(n)`

Reply:
Huh, I didn't realize (or forgot?) that numpy datatypes didn't pass basic isinstance checks, though it makes sense.

After reading through the numbers docs: https://docs.python.org/3/library/numbers.html
- All of these numbers are instances of Real (including numpy floats)
- numpy ints and built-in ints are instances of Integral

So the generic "is this an integer" maps to `isinstance(my_number, numbers.Integral)`, and the generic "is this a float" maps to `isinstance(my_number, numbers.Real) and not isinstance(my_number, numbers.Integral)`?

That's pretty terrible if true

Reply:
I guess it comes back to the base idea here being really awkward and unintuitive to the uninitiated: there is no reason for programs to differentiate between whether or not the user has given exactly a float in most cases, since we expect 1 and 1.0 to do the same thing in general, so the standard library isn't built around these sorts of use cases.